internal/controllers/clusterrequest_controller.go

```diff
@@ -612,59 +646,77 @@ func (t *clusterRequestReconcilerTask) handleClusterInstallation(ctx context.Con
 		existingClusterInstance)
 	if err != nil {
 		if !errors.IsNotFound(err) {
-			return fmt.Errorf("failed to get ClusterInstance: %w", err)
+			return nil, fmt.Errorf("failed to get ClusterInstance: %w", err)
+		}
+
+		operationType := "created"
+		opts := []client.CreateOption{}
+		if isDryRun {
+			opts = append(opts, client.DryRunAll)
+			operationType = "validated with dry-run"
 		}
 
 		// Create the ClusterInstance
-		err = t.client.Create(ctx, clusterInstance)
+		err = t.client.Create(ctx, clusterInstance, opts...)
 		if err != nil {
 			if !errors.IsInvalid(err) && !errors.IsBadRequest(err) {
-				return fmt.Errorf("failed to create ClusterInstance: %w", err)
+				return nil, fmt.Errorf("failed to create ClusterInstance: %w", err)
 			}
 			// Invalid or webhook error
-			err = utils.NewInputError(err.Error())
-			t.updateClusterInstanceProcessedStatus(nil, err)
-		} else {
-			t.logger.InfoContext(
-				ctx,
-				fmt.Sprintf(
-					"Created ClusterInstance %s in the namespace %s",
-					clusterInstance.GetName(),
-					clusterInstance.GetNamespace(),
-				),
-			)
+			return nil, utils.NewInputError(err.Error())
```

Comment:
Hmm, why was the call to `t.updateClusterInstanceProcessedStatus(nil, err)` removed? :thinking: